src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinder.cs

```diff
@@ -705,11 +705,8 @@ private Expr CreateArgumentEXPR(ArgumentObject argument, LocalVariableSymbol loc
             // as well so that overload resolution can find them.
             if (callingType.IsWindowsRuntimeType())
             {
-                TypeArray collectioniFaces = callingType.GetWinRTCollectionIfacesAll(SymbolLoader);
```

Comment:
Ha! I just came online to close this as I'd come to a similar conclusion. I'd started experimenting with making this generic so it could be more strongly typed, and quickly decided that just using arrays and lists would be better all-round. This class is the worse of both worlds between those two.
In the meantime this offers too temporary a minor improvement, considering it likely conflicts with at least one of the other PRs I have open.